api/handlers/zip_package.go

```diff
@@ -20,6 +20,16 @@ type ZippedPackage struct {
 	extractedDir string
 }
 
+func (m *ZippedPackage) GetResource(resourceName string) (io.ReadCloser, error) {
+	actualPath := path.Join(m.extractedDir, resourceName)
+	file, err := os.Open(actualPath)
+	if err != nil {
+		return nil, fmt.Errorf("error accessing resource %s: %w", resourceName, err)
+	}
+
+	return file, nil
+}
```

Comment:
Not sure if it applies to this piece of code, but could one hypothetically abuse this by, let's say, calling `GetResource("../../../../../../etc/passwd")`?

In any case, we could try to make sure that we are not allowing to go above the base directory - not sure how to do that in Go though...

Reply:
We can:
-  Clean() the path and then check if we are still inside the temp directory
- Forbid the `.` and `..` in the manifest.yaml references
- Not extract any file and process everything by reading the zip file directly

Reply:
> Not extract any file and process everything by reading the zip file directly
That would be my favorite, but let's not go that route for the Pull Request.

IMO cleaning the path, and then using `strings.HasPrefix` or a similar methodology is sufficient for now. Not sure if this is actually a problem, or if it's just me being paranoid.

Reply:
@christian-kreuzberger-dtx  @Raffy23  Please have another look at the latest fixup

Reply:
I think the current implementation is sufficient to prevent most path traversal attacks, but I haven't looked too much into it.